apps/web/components/ForYou.tsx

```diff
@@ -0,0 +1,129 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import { getEventContract } from "@/lib/getEventContract";
+import { getEventFactoryContract } from "@/lib/getEventFactoryContract";
+import { ContractPermission, EventSea } from "@/types";
+
+const ForYou: React.FC = () => {
+  const [selectedCountry, setSelectedCountry] = useState("🇺🇸 USA");
+  const [events, setEvents] = useState<EventSea.Event[]>([]); // State for events
+
+  // Function to fetch events from the blockchain
+  const fetchEvents = async () => {
```

Comment:
why do we need to fetch events here again ? can we pass events as a prop from home page and do the filtering in this component ?

Reply:
Thank you for pointing it out, I will try the method as you mentioned. @chin-flags